### PR TITLE
update weights optimization

### PR DESCRIPTION
**Pull request by clement-pages**
BREAKING: remove `finetune_wavlm` attributes from the `PixIT` task, and replace it with `walm_frozen` in `Totatonet` separation model, to be consistent with what is done for the segmentation model.

### CODE REVIEW

**Comment by clement-pages** [comment]
Merging this PR should fix #1789

**Comment by hbredin** [comment]
@Lebourdais suggested that we set `wavlm.requires_grad` to False in `__init__` instead of using the `with torch.no_grad` logic.  I think it is indeed a better idea as it allows to get a better estimate of the number of trainable parameters (i.e. those for which `requires_grad` is set to True.

@clement-pages can you update your PR to make this change and also update the `SSeRiouSS` model and diarization task accordingly?

**Comment by clement-pages** [comment]
> @clement-pages can you update your PR to make this change and also update the SSeRiouSS model and diarization task accordingly?

OK, I will update it.

File: pyannote/audio/tasks/segmentation/speaker_diarization.py [hunk 1]
```diff
@@ -467,6 +467,9 @@ def training_step(self, batch, batch_idx: int):
 
         if not self.automatic_optimization:
             optimizers = self.model.optimizers()
+            optimizers = (
+                [optimizers] if not isinstance(optimizers, list) else optimizers
+            )
```
Comment (hbredin) [inline pyannote/audio/tasks/segmentation/speaker_diarization.py:472]:
```suggestion
            optimizers = optimizers if isinstance(optimizers, list) else [optimizers]
```

File: pyannote/audio/tasks/separation/PixIT.py [hunk 2]
```diff
@@ -1012,6 +1012,9 @@ def training_step(self, batch, batch_idx: int):
         # using multiple optimizers requires manual optimization
         if not self.automatic_optimization:
             optimizers = self.model.optimizers()
+            optimizers = (
+                [optimizers] if not isinstance(optimizers, list) else optimizers
+            )
```
Comment (hbredin) [inline pyannote/audio/tasks/separation/PixIT.py:1017]:
```suggestion
            optimizers = optimizers if isinstance(optimizers, list) else [optimizers]
```

**Review by hbredin — CHANGES_REQUESTED** [review]

File: pyannote/audio/models/separation/ToTaToNet.py [hunk 3]
```diff
@@ -137,6 +139,7 @@ def __init__(
         task: Optional[Task] = None,
         n_sources: int = 3,
         use_wavlm: bool = True,
+        wavlm_frozen: bool = False,
```
Comment (hbredin) [inline pyannote/audio/models/separation/ToTaToNet.py:142]: Did you double check that it does not break loading https://huggingface.co/pyannote/separation-ami-1.0?

Comment (clement-pages) [reply pyannote/audio/models/separation/ToTaToNet.py:142]: No, I will check it tomorrow.

**Pull request merged**